projects/storefrontlib/src/lib/cms/store/effects/page.effect.ts

```diff
@@ -125,8 +134,8 @@ export class PageEffects {
     }
   }
 
-  private getComponents(pageData: any) {
-    const components: any[] = [];
+  private getComponents(pageData: CMSPage) {
+    const components: Component[] = [];
```

Comment:
use `<T extends Component>`, and add return type.

Reply:
The `component` const in  `const component of slot.components.component` is of type `Component` and thus cannot be downcasted to a type of `T`, where `T` extends `Component`

Reply:
I prefer to use `any` rather than `CmsComponent`. ContentSlot can contain any types of components.